kpi/deployment_backends/kc_reader/utils.py

```diff
@@ -0,0 +1,68 @@
+from django.contrib.auth.models import User
+from django.db import models
+from django.db import ProgrammingError
+
+
+class __ReadOnlyModelError(ValueError):
+    pass
+
+
+class __ReadOnlyModel:
```

Comment:
There are other methods that write, e.g. `update()`, but this is a good safety net for most accidents